nuxeo-core/nuxeo-core-mongodb/src/main/java/org/nuxeo/ecm/core/mongodb/seqgen/MongoDBUIDSequencer.java

```diff
@@ -65,6 +69,21 @@ public void init() {
         getSequencerCollection();
     }
 
+    @Override
+    public void initSequence(String key, int id) {
+        try {
+            Bson filter = and(eq(MongoDBSerializationHelper.MONGODB_ID, key), not(gte(SEQUENCE_VALUE_FIELD, id)));
+            Document sequence = new Document();
+            sequence.put(MongoDBSerializationHelper.MONGODB_ID, key);
+            sequence.put(SEQUENCE_VALUE_FIELD, id);
+            getSequencerCollection().replaceOne(filter, sequence, new UpdateOptions().upsert(true));
+        } catch (MongoWriteException e) {
+            if (log.isWarnEnabled()) {
+                log.warn("Failed to update the sequence '" + key + "' with value "+id, e);
+            }
```

Comment:
If there's a problem you must not log but rethrow an exception (NuxeoException). Logging would mean things will usually get buried, and that's not what we want; we want to crash if something is wrong with the MongoDB access or the sequence value.